bench/run_bench.py

```
@@ -0,0 +1,478 @@
+
+        The binary being on PATH is not enough: without PMU access (GCE without
+        vPMU, or a strict `kernel.perf_event_paranoid`) perf runs fine and
+        reports `<not supported>` for every event. Selecting the backend on
+        `which perf` alone made `counters_available()` lie, so the
+        graceful-degradation path never engaged and the run died on the first
+        measurement instead of reporting instr/cycles as absent.
+        """
+        if not _PERF:
+            return False
+        try:
+            out = subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)